File: 844-backspace-string-compare/844-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        stack<char> a, b;
        
        for(char &c: s){
            if(c == '#' && !a.empty()) 
                a.pop();
            else if(c != '#')
                a.push(c);
        }
        
        for(char &c: t){
            if(c == '#' && !b.empty())
                b.pop();
            else if(c != '#')
                b.push(c);
        }
        
        string ss, tt;
        while(!a.empty())
            ss.push_back(a.top()), a.pop();
        
        while(!b.empty())
            tt.push_back(b.top()), b.pop();
        
        cout << ss << " " << tt << endl;
        return ss == tt;
    }
};

File: 844-backspace-string-compare/844-backspace-string-compare.cs
public class Solution {
    public bool BackspaceCompare(string s, string t) {
        return Backspaced(s) == Backspaced(t);
    }
    
    public string Backspaced(string s){
        int n = s.Length;
        Stack<char> st = new Stack<char>();
        foreach(char c in s){
            if(c == '#'){
                if(st.Count > 0)
                    st.Pop();
            }
            else{
                st.Push(c);
            }
        }
        char[] lc = st.ToArray();
        return new string(lc);
    }
}